net/golioth/system_client.c

```diff
@@ -454,11 +452,19 @@ K_THREAD_DEFINE(golioth_system, CONFIG_GOLIOTH_SYSTEM_CLIENT_STACK_SIZE,
 
 void golioth_system_client_start(void)
 {
+	int ret = 0;
 	if (IS_ENABLED(CONFIG_GOLIOTH_AUTH_METHOD_PSK)) {
 		golioth_settings_check_credentials();
+	} else if (IS_ENABLED(CONFIG_GOLIOTH_AUTH_METHOD_CERT)) {
+		ret = golioth_check_cert_credentials();
+	}
+
+	if (ret == 0) {
+		k_sem_give(&sys_client_started);
+	} else {
+		LOG_WRN("Error loading TLS credentials, golioth system client was not started");
```

Comment:
```suggestion
	int err = 0;

	if (IS_ENABLED(CONFIG_GOLIOTH_AUTH_METHOD_PSK)) {
		golioth_settings_check_credentials();
	} else if (IS_ENABLED(CONFIG_GOLIOTH_AUTH_METHOD_CERT)) {
		err = golioth_check_cert_credentials();
	}

	if (err) {
		LOG_WRN("Error loading TLS credentials, golioth system client was not started");
	} else {
		k_sem_give(&sys_client_started);
```

Reply:
I prefer to have the happy path first, and error handling in the `else` clause. I'll update to the `err` naming to match the rest of the file.